### ISSUE TITLE
Edge function does not execute in parallel

### ISSUE THREAD

**Issue by candymandev** [supabase/supabase#7253]
I have an edge function `generate-image`. When I invoke the function locally many times, each invocation runs in parallel as I expect.

![local](https://user-images.githubusercontent.com/94403944/173210070-dab51903-123a-40fa-b67e-4a4598731e2f.png)

When I deploy the function to production with the same code to invoke it many times, it seems to run sequentially. Something on the server side seems to be limiting the function to run only one execution at a time.

![production](https://user-images.githubusercontent.com/94403944/173210091-9b968ed8-4ae1-4bd1-ad37-f8f857718665.png)

According to the function metrics in Supabase, the function executes in less than a second. The duration of the invocations is pretty consistent. So it seems that although the HTTP requests are initiated properly, but something on the server side is causing each subsequent invocation to start only when the last one completes.

My expectation is that these functions would run completely in parallel up to the limits of my pro plan. Having these functions run sequentially misses out on one of the biggest benefits of serverless edge functions in my opinion. is this a limitation of my plan? Is there a way to improve the performance of my function?

**Comment by candymandev** [comment]
For comparison, here is the same edge function running on Deno Deploy. The invocations execute in parallel as I expect.

![screenshot-2022-06-13-11-21-40](https://user-images.githubusercontent.com/94403944/173388136-481bdcde-e19e-4965-ad53-cfb35ed45366.png)

They take much longer and I'm hitting memory limits on Deno Deploy when I do a lot of these... but that's another issue altogether :joy:
